src/System.Private.CoreLib/src/System/Utf8String.Comparison.cs

```diff
@@ -0,0 +1,142 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Text;
+
+namespace System
+{
+    public sealed partial class Utf8String
+    {
+        /*
+         * COMPARISON OF UTF-8 AGAINST UTF-16
+         */
+
+        /// <summary>
+        /// Returns a value stating whether <paramref name="utf8Text"/> and <paramref name="utf16Text"/>
+        /// represent the same data. An ordinal comparison is performed scalar-by-scalar.
+        /// </summary>
+        /// <remarks>
+        /// This method returns <see langword="true"/> if both <paramref name="utf8Text"/> and
+        /// <paramref name="utf16Text"/> are null, or if both are empty. This method returns <see langword="false"/>
+        /// if either input contains an ill-formed subsequence. Otherwise, this method returns <see langword="true"/>
+        /// if and only if both arguments decode to the same Unicode scalar value sequence.
+        /// </remarks>
+        public static bool AreEquivalent(Utf8String? utf8Text, string? utf16Text)
```

Comment:
>AreEquivalent [](start = 27, length = 13)

what not naming this Equals?

Reply:
Conflicts with the `object.Equals(object, object)` overload, mainly. I didn't want devs to get into the habit of calling `ustring.Equals`, accidentally hitting the wrong overload because one of the parameters was typed incorrectly, and see unexpected failures. But TBH I'm not very hard-line on this idea, so we can reconsider it during API review. :)